Sources/GRPC/ClientOptions.swift

```diff
@@ -60,24 +61,33 @@ public struct CallOptions {
   /// messages associated with the call.
   public var requestIDHeader: String?
 
+  /// A logger used for the call. Defaults to a no-op logger.
+  ///
+  /// If a `requestIDProvider` exists then a request ID will automatically attached to the logger's
+  /// metadata using the 'grpc-request-id' key.
+  public var logger: Logger
+
   public init(
     customMetadata: HPACKHeaders = HPACKHeaders(),
     timeLimit: TimeLimit = .none,
     messageEncoding: ClientMessageEncoding = .disabled,
-    requestIDProvider: RequestIDProvider = .autogenerated,
+    requestIDProvider: RequestIDProvider = .none,
```

Comment:
What's the reason for this change?

Reply:
It seemed more natural to opt-in to having request IDs generated rather than opting out. Not sure how much I agree with that thinking now. WDYT?

Reply:
I'm inclined to want to go to "fall into a pit of success": if users don't express a preference, giving them nice UUIDs makes their lives easier, and also our own during bug triage.